### PR TITLE
Added captions to lesson videos

### LINKED ISSUE #71: Captions on lesson videos
Load captions available in [sysassets](https://github.com/angrave/sysassets) and show them as the lesson video is playing.

More information to come.

### PR DESCRIPTION
**Pull request by scowalt**
This resolves #71.

Obviously this was a bit of a headache to put together. Here are some notes:

* Since `sysassets` and `sysbuild` are both in the `cs-education.github.io` subdomain, the xdomain library is no longer necessary. See the current deployment on [sys-staging](http://cs-education.github.io/sys-staging) to see this in action.
* There currently isn't a great way to have a remote text track be automatically selected after it's added. Thus, the CSS hack in `router.js:141`.
* There also isn't a great way to modify the location of text tracks using CSS in videojs. The library automatically adjusts around CSS hacks whenever a new caption is displayed. Seems like the best way put the text track outside of the video would be to either modify the videojs library or use a external captioning library. From my experience, however, the concern of captions blocking a code example on the screen was not an issue.

Some of the issues I had could be resolved by using a custom caption library like [videojs-caption](http://sampingchuang.com/videojs-caption) or [classTranscribe](https://github.com/cs-education/classTranscribe). However, I feel that using a defined web standard like WebVTT contributes to the sustainability and accessibility of the project. Hopefully the videojs library will add some features going forward to make our lives easier.

**TL;DR** Captions are done

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -325,6 +325,7 @@ <h4 class="modal-title" id="gcc-error-label" data-bind="html: errorWarningLabel"
         <!-- build:js scripts/vendor.js -->
         <!-- bower:js -->
         <script src="../bower_components/jquery/dist/jquery.js"></script>
+        <script src="../bower_components/bootstrap-sass-official/assets/javascripts/bootstrap.js"></script>
```
Comment (neelabhg) [inline app/index.html]: Was this added by `grunt bowerInstall`? The Bootstrap JS files are already included as `scripts/plugins.js` in a build block further down in this page, so I don't think this line is needed.

**Comment by neelabhg** [comment]
Removing xdomain makes sense, but whats the harm in keeping it? Maybe later down the line we would need it.

File: app/index.html (hunk 1 above)
Comment (scowalt) [inline app/index.html]: That change (and the other jquery.notific8 one) keeps getting made automatically by something. This also happens to @jdtran23. I really don't know anything else about it, but I can remove the change if need be

**Comment by scowalt** [comment]
If we keep in imported in `index.html` and configured the way it is, it will intercept requests going accross `cs-education.github.io`. This is fine in theory, but in reality there is an issue loading text tracks using this method in Chrome (something about the data type being transmitted as an octet-stream but interpreted as WebVTT (this is demonstrated in a previous iteration of sys-staging)). Thus, if we wanted to keep the library included in the project, we would have to disable it in order to achieve the same level of functionality.

File: app/index.html (hunk 1 above)
Comment (neelabhg) [inline app/index.html]: Probably added by `bowerInstall` when watching for changes. Its okay, I'll remove those myself once this is merged into `master`.

**Comment by neelabhg** [comment]
I see. The other things look good to me, though I think merging @jdtran23 's PR first makes sense otherwise he'll have to fix merge conflicts later.

**Comment by scowalt** [comment]
I think that makes sense. I can expand this pull request to include my refactoring of his module as well.

**Comment by neelabhg** [comment]
Please do. And merge conflicts AAAAAA!

**Comment by scowalt** [comment]
Alright, I fixed all of the conflicts. However, I would prefer to wait until I can do some refactoring before merging this in.

**Comment by scowalt** [comment]
Please don't merge this change quite yet; subtitles still aren't working on Chrome

**Comment by scowalt** [comment]
Alright this is working now, video will still play if subtitles don't load, check out sys-staging for a working example.

**Comment by neelabhg** [comment]
Looks good to me. The subtitles work very well in staging. Great work!

**Comment by angrave** [comment]
On 4/25/15 4:28 PM, Neelabh Gupta wrote:
>
> Looks good to me. The subtitles work very well in staging. Great work!
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/cs-education/sysbuild/pull/97#issuecomment-96281193>.
>
Can we push this out to prod?

**Comment by bobrenjc93** [comment]
Awesome :+1: :ship:
